images/cilium/Dockerfile

```diff
@@ -0,0 +1,38 @@
+# syntax=docker/dockerfile:1.1-experimental
+ARG CILIUM_BUILDER_IMAGE=docker.io/errordeveloper/cilium-builder:019538f3bd3275e8d8369e2c5823efbe0f3a4f06-dev
+ARG CILIUM_RUNTIME_IMAGE=docker.io/errordeveloper/cilium-runtime:201594d798c0dafdc5969e81ca21fa2761103b7f-dev
+
+#FROM quay.io/cilium/hubble:v0.6.0 as hubble
+
+FROM --platform=linux/amd64 ${CILIUM_BUILDER_IMAGE} as builder
+
+ARG NOSTRIP
+ARG LOCKDEBUG
+
+RUN --mount=type=bind,readwrite,target=/go/src/github.com/cilium/cilium --mount=target=/root/.cache,type=cache --mount=target=/go/pkg/mod,type=cache \
+  make clean-container build-container install-container \
+    NOSTRIP=$NOSTRIP LOCKDEBUG=$LOCKDEBUG PKG_BUILD=1 SKIP_DOCS=true DESTDIR=/out/linux/amd64
+
+RUN --mount=type=bind,readwrite,target=/go/src/github.com/cilium/cilium --mount=target=/root/.cache,type=cache --mount=target=/go/pkg/mod,type=cache \
+  env GOARCH=arm64 CC=aarch64-linux-gnu-gcc \
+    make clean-container build-container install-container \
+      NOSTRIP=$NOSTRIP LOCKDEBUG=$LOCKDEBUG PKG_BUILD=1 SKIP_DOCS=true DESTDIR=/out/linux/arm64 \
+      HOST_CC=aarch64-linux-gnu-gcc HOST_STRIP=aarch64-linux-gnu-strip
+
+FROM ${CILIUM_RUNTIME_IMAGE}
+ARG TARGETPLATFORM
+LABEL maintainer="maintainer@cilium.io"
+
+COPY --from=builder /out/${TARGETPLATFORM} /
+
+#COPY --from=hubble /usr/bin/hubble /usr/bin/hubble
+#COPY --from=builder /go/src/github.com/cilium/cilium/plugins/cilium-cni/cni-install.sh /cni-install.sh
+#COPY --from=builder /go/src/github.com/cilium/cilium/plugins/cilium-cni/cni-uninstall.sh /cni-uninstall.sh
+#COPY --from=builder /go/src/github.com/cilium/cilium/contrib/packaging/docker/init-container.sh /init-container.sh
+
```

Comment:
Where did `cilium/cilium-envoy` end up?

Reply:
It requires more work, I've managed to compile Envoy itself with a few work-arounds, but making it happen for proxy is going to take a few days. From discussion with @aanm It seems reasonable to omit L7 support from the first version.

Reply:
I think that's fine as long as we're clear about the limitations and any workarounds necessary to make that work. For instance, I believe that on startup Cilium will check the envoy binary availability and version to ensure it's compatible. I'm pretty sure there's a flag to override this, but what's the flag? etc.